File: src/config/type/JsonConfigType.php
<?php

namespace pocketcloud\cloud\config\type;

use JsonException;

final class JsonConfigType implements ConfigType {

    /** @throws JsonException */
    public function decodeContent(string $content): array {
        return json_decode($content, true, flags: JSON_THROW_ON_ERROR);
    }

    /** @throws JsonException */
    public function encodeContent(array $content): string {
        return json_encode($content, flags: JSON_THROW_ON_ERROR | JSON_PRETTY_PRINT | JSON_BIGINT_AS_STRING | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
    }
}

File: src/main/java/de/pocketcloud/cloud/config/type/JsonConfigType.java
package de.pocketcloud.cloud.config.type;

import de.pocketcloud.cloud.util.FileUtils;

import java.util.Map;

public final class JsonConfigType implements ConfigType {

    @Override
    @SuppressWarnings("unchecked")
    public Map<String, Object> decode(String content) {
        return FileUtils.decodeJson(content, Map.class);
    }

    @Override
    public String encode(Map<String, Object> content) {
        return FileUtils.encodeJson(content);
    }
}